types/mochawesome/addContext.d.ts

```diff
@@ -0,0 +1,35 @@
+import * as Mocha from 'mocha';
+
+interface TestContextObject {
+    title: string;
+    value: any;
+}
+
+type TestContext = string | TestContextObject;
+
+/**
+ * Add context to the test object so it can
+ * be displayed in the mochawesome report.
+ *
+ * **Note**: Arrow function tests are not currently
+ * supported, since we depend on the current test
+ * context (i.e. `this`)
+ *
+ * @example
+ * it('should test something', function () {
+ *   someFunctionThatTestsCode();
+ *
+ *   addContext(this, 'some context to add');
+ *
+ *   addContext(this, {
+ *     title: 'Expected number of something'
+ *     value: 42
+ *   });
+ *
+ *   assert('something');
+ * });
+ *
+ */
+declare function addContext(testCtx: Mocha.Context, context: TestContext): void;
```

Comment:
You should wrap it all in a `namespace` for the type declarations to become importable:
```suggestion
declare namespace addContext {
    interface TestContextObject {
        title: string;
        value: any;
    }

    type TestContext = string | TestContextObject;
}

/**
 * Add context to the test object so it can
 * be displayed in the mochawesome report.
 *
 * **Note**: Arrow function tests are not currently
 * supported, since we depend on the current test
 * context (i.e. `this`)
 *
 * @example
 * it('should test something', function () {
 *   someFunctionThatTestsCode();
 *
 *   addContext(this, 'some context to add');
 *
 *   addContext(this, {
 *     title: 'Expected number of something'
 *     value: 42
 *   });
 *
 *   assert('something');
 * });
 *
 */
declare function addContext(testCtx: Mocha.Context, context: addContext.TestContext): void;
```

Reply:
I agree with this, thanks for the suggestion. Updating now.